### PR TITLE
Parameterize `CheckRules` and `ValidateRule`

### PR DESCRIPTION
**Pull request by joneshf**
Might solve #3?

From discussions in slack.

### CODE REVIEW

File: src/HomeRunBall.purs [hunk 1]
```diff
@@ -16,7 +16,7 @@ import Type.Row (Cons, Nil, kind RowList)
 -- | Check a string for validation rules provided by a row proxy and return a validation result
 checkRules :: forall row errors rl
```
Comment (justinwoo) [inline src/HomeRunBall.purs]: should be updated to take `a` and allow the String to be replaced by `a`?

Comment (justinwoo) [reply src/HomeRunBall.purs]: oh, sorry, realized you were talking about this specific thing. let me try to read more...

Comment (justinwoo) [reply src/HomeRunBall.purs]: btw this should be updated along with type VS becoming VV, https://github.com/justinwoo/purescript-home-run-ball/pull/5/files#diff-6577c4f181007b0640ffddc1d6f6f7fcR44

**Comment by justinwoo** [comment]
Mind if I merge this?

**Comment by joneshf** [comment]
Do it to it.

**Comment by justinwoo** [comment]
thanks!!!

**Pull request merged**